torchx/schedulers/test/ray_scheduler_test.py

```diff
@@ -298,6 +298,18 @@ def test_requirements(self) -> None:
                 job = req.request
                 self.assertEqual(job.requirements, reqs)
 
+        def test_parse_app_id(self) -> None:
+            from itertools import product
+            addrs = ["0.0.0.0", "www.testexample.com", "addr-of-cluster"]
+            ports = ["65535", ]
+            app_ids = ["a-p-p_id", "app-i-d", "app_id", "app:id"]
+            for addr, port, app_id in product(addrs, ports, app_ids):
```

Comment:
product is clever but I think it's a bit easy to miss things since it's a bit hard to follow

no strong preference but might be better to just be very explicit and call _parse_app_id for each case even if it's a few more lines of code

Reply:
OK, I have changed the test examples to explicit form.